# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/GlobalObject.cpp
# Commit: 6f7514b0a657
# Full Hash: 6f7514b0a657f97bf3e0e097c2b3faaea142450b
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2021-03-06 09:33:16
# Regressor Bug: 1689413
# File Overlap Count: 1
# Description:
#   Bug 1689413 part 2 - Remove owned BaseShapes. r=jonco
#   
#   At this point all BaseShapes are unowned (= immutable, shared) so we can remove the
#   UnownedBaseShape type and the `BaseShape::unowned_` field.
#   
# ==============================================================================

diff -r 76db3b04dedd -r 6f7514b0a657 js/src/vm/GlobalObject.cpp
--- a/js/src/vm/GlobalObject.cpp	Fri Mar 05 19:10:04 2021 +0000
+++ b/js/src/vm/GlobalObject.cpp	Fri Mar 05 19:10:05 2021 +0000
@@ -1079,7 +1079,7 @@
 
   uint32_t slot = holder->slotSpan();
   RootedShape last(cx, holder->lastProperty());
-  Rooted<UnownedBaseShape*> base(cx, last->base()->unowned());
+  Rooted<BaseShape*> base(cx, last->base());
 
   RootedId id(cx, NameToId(name));
   Rooted<StackShape> child(cx, StackShape(base, id, slot, 0));